airbyte-integrations/connectors/source-looker/source_looker/client.py

```diff
@@ -84,11 +137,84 @@ def _request(self, url: str, method: str = "GET", data: dict = None) -> List[dic
                 return [response_data]
         return []
 
+    def _get_run_look_key(self, look_id, look_name):
+        return f"{look_id} - {look_name}"
+
+    def _get_explore_fields(self, model, explore):
+        """
+        For a given LookML model and explore, looks up its dimensions/measures
+        and their types for run_look endpoint JSON Schema generation
+        """
+        if (model, explore) not in self._run_look_explore_fields:
+            self._run_look_explore_fields[(model, explore)] =  self._request(f"{self.BASE_URL}/lookml_models/{model}/explores/{explore}?fields=fields(dimensions(name,type),measures(name,type))")[0]['fields']
+        
+        return self._run_look_explore_fields[(model, explore)]
+
+
+    def _get_look_field_schema(self, model, field):
+        """
+        For a given LookML model and field, looks up its type and generates
+        its properties for the run_look endpoint JSON Schema
+        """
+        explore = field.split(".")[0]
+        
+        fields = self._get_explore_fields(model, explore)
+
+        field_type = "string" # default to string
+        for dimension in fields['dimensions']:
+            if field == dimension['name'] and dimension['type'] in self._field_type_mapping:
+                field_type = self._field_type_mapping[dimension['type']]
+        for measure in fields['measures']:
+            if field == measure['name']:
+                # Default to number except for list, date, and yesno
+                field_type = "number"
+                if measure['type'] in self._field_type_mapping:
+                    field_type = self._field_type_mapping[measure['type']]
+
+        if field_type == 'datetime':
+            # no datetime type for JSON Schema
+            return {
+                "type": ["null", "string"],
+                "format": "date-time"
+            }
+        else:
+            return {
+                "type": ["null", field_type]
+            }
+        
+        
+    def _get_run_look_json_schema(self):
```

Comment:
nit: can we order these functions in the order in which they are called from the main? e.g. _get_run_look_json_schema should be earlier than _get_look_field_schema since the former is called first and calls the latter. I'm dizzy from jumping around :P